### PR TITLE
workflows: use `set -xeuo pipefail` globally

### PR DESCRIPTION
**Pull request by carlocab**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [ ] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

This makes it easier to avoid subtle shell script bugs.

### CODE REVIEW

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -41,7 +45,9 @@ jobs:
           date="$(date --rfc-3339=seconds --utc)"
           brew_version="$(git describe --tags --dirty --abbrev=7)"
 
+          set +o pipefail
           DELIMITER="END_LABELS_$(LC_ALL=C tr -dc '[:alnum:]' </dev/urandom | head -c20)"
+          set -o pipefail
```
Comment (Bo98) [inline .github/workflows/docker.yml:50]: What's this one about?

**Review by Bo98 — APPROVED** [review]

File: .github/workflows/docker.yml (hunk 1 above)
Comment (Bo98) [reply .github/workflows/docker.yml:50]: Or perhaps better: use `uuidgen` or similar.

**Comment by Bo98** [comment]
Looks like there's another delimiter generator later on in the step that you're perhaps not hitting in CI but might do in a release workflow.

**Review by MikeMcQuaid — APPROVED** [review]
Love it (although it seems familiar from somewhere, can't think of where 😉)!

**Comment by MikeMcQuaid** [comment]
@carlocab good to self-merge whenever you're 👍🏻 as CI is 🟢

**Comment by carlocab** [comment]
Yes, will do start of tomorrow so I can be sure to be around if something goes :boom:

**Comment by fiftydinar** [comment]
`set -euxo pipefail` is more correct to use.
Using incorrect set order can potentially result in undefined error log in some situations when using command pipe.

See
https://www.gnu.org/software/bash/manual/html_node/The-Set-Builtin.html

**Pull request merged**